internal/discovery/foreign-cluster-operator/foreign-cluster-operator_test.go

```diff
@@ -642,4 +636,245 @@ var _ = Describe("ForeignClusterOperator", func() {
 
 	})
 
+	Context("Test Permission", func() {
+
+		const (
+			outgoingBinding = "liqo-binding-liqo-outgoing"
+			incomingBinding = "liqo-binding-liqo-incoming"
+		)
+
+		var (
+			clusterRole1 rbacv1.ClusterRole
+			clusterRole2 rbacv1.ClusterRole
+		)
+
+		type permissionTestcase struct {
+			fc               discoveryv1alpha1.ForeignCluster
+			expectedOutgoing types.GomegaMatcher
+			expectedIncoming types.GomegaMatcher
+		}
+
+		JustBeforeEach(func() {
+			clusterRole1 = rbacv1.ClusterRole{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "liqo-outgoing",
+				},
+			}
+			clusterRole2 = rbacv1.ClusterRole{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "liqo-incoming",
+				},
+			}
+
+			Expect(controller.Client.Create(ctx, &clusterRole1)).ToNot(HaveOccurred())
+			Expect(controller.Client.Create(ctx, &clusterRole2)).ToNot(HaveOccurred())
+
+			controller.peeringPermission = peeringroles.PeeringPermission{
+				Basic: []*rbacv1.ClusterRole{},
+				Incoming: []*rbacv1.ClusterRole{
+					&clusterRole2,
+				},
+				Outgoing: []*rbacv1.ClusterRole{
+					&clusterRole1,
+				},
+			}
+		})
+
+		JustAfterEach(func() {
+			var roleBindingList rbacv1.RoleBindingList
+			Expect(controller.Client.List(ctx, &roleBindingList)).ToNot(HaveOccurred())
+			for i := range roleBindingList.Items {
+				rb := &roleBindingList.Items[i]
+				Expect(controller.Client.Delete(ctx, rb)).ToNot(HaveOccurred())
+			}
```

Comment:
Nit. Could be probably replaced by a `controller.Client.DeleteAllOf(...)` call

Reply:
No, it tells me that the `deletecollection` verb is not allowed by this API Server on that resource

Reply:
Ah ok, didn't know about that